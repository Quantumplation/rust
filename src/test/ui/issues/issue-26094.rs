macro_rules! some_macro {
    ($other: expr) => ({
        $other(None) //~ NOTE supplied 1 argument
    })
}

fn some_function() {} //~ NOTE defined here

fn main() {
    some_macro!(some_function);
    //~^ ERROR arguments to this function are incorrect
    //~| NOTE argument unexpected
}
