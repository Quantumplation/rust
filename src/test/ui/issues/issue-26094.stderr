error[E0308]: arguments to this function are incorrect
  --> $DIR/issue-26094.rs:3:9
   |
LL |         $other(None)
   |         ^^^^^^^----^
   |                |
   |                argument unexpected
...
LL |     some_macro!(some_function);
   |     --------------------------- in this macro invocation
   |
note: function defined here
  --> $DIR/issue-26094.rs:7:4
   |
LL | fn some_function() {}
   |    ^^^^^^^^^^^^^
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: remove the extra argument
   |
LL |         some_function()
   |         ^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
